### PR TITLE
Clean up yang parse tree after #118

### PR DESCRIPTION
**Pull request by pudelkoM**
Found a much better and cleaner solution.

### CODE REVIEW

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 1]
```diff
@@ -371,43 +379,48 @@ U GetValue(
...
-
-  // Retrieve value from the switch.
+  DataResponseWriter writer(
+      [&resp, data_response_get_inner_message_func,
+       data_response_has_inner_message_func,
+       optical_channel_info_has_inner_message_func,
+       optical_channel_info_get_inner_message_func,
+       inner_message_get_field_func](const DataResponse& in) {
+        if (!(in.*data_response_has_inner_message_func)()) return false;
+        auto oc_info = (in.*data_response_get_inner_message_func)();
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc]: `oc_info` ->`data_req_inner_msg` ?

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 2]
```diff
@@ -371,43 +379,48 @@ U GetValue(
 }
 
 // Optical-port-specific version.
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc]: This becomes a general function. I think we can rename some variable

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 3]
```diff
@@ -371,43 +379,48 @@ U GetValue(
...
+U GetValue(
     uint64 node_id, uint32 port_id, YangParseTree* tree,
-    const T& (OpticalChannelInfo::*instance_getter)() const,
-    bool (OpticalChannelInfo::*has_instance_getter)() const,
-    U (T::*inner_message_get_field_func)() const) {
+    const T& (DataResponse::*data_response_get_inner_message_func)() const,
+    bool (DataResponse::*data_response_has_inner_message_func)() const,
+    DataRequest::Request::Port* (
+        DataRequest::Request::*data_request_get_mutable_inner_message_func)(),
+    bool (T::*optical_channel_info_has_inner_message_func)() const,
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc]: `data_req_has_inner_message_func` ?

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 4]
```diff
@@ -371,43 +379,48 @@ U GetValue(
...
+    const V& (T::*optical_channel_info_get_inner_message_func)() const,
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc]: `data_req_get_inner_message_func`?

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 5]
```diff
@@ -371,43 +379,48 @@ U GetValue(
...
+      [&resp, data_response_get_inner_message_func,
+       data_response_has_inner_message_func,
+       optical_channel_info_has_inner_message_func,
+       optical_channel_info_get_inner_message_func,
+       inner_message_get_field_func](const DataResponse& in) {
+        if (!(in.*data_response_has_inner_message_func)()) return false;
+        auto oc_info = (in.*data_response_get_inner_message_func)();
+        if (!(oc_info.*optical_channel_info_has_inner_message_func)())
+          return false;
+        auto power = (oc_info.*optical_channel_info_get_inner_message_func)();
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc]: `power` -> `data_req_inner_msg_field` ? (or something else

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 6]
```diff
@@ -525,6 +538,30 @@ TreeNodeEventHandler GetOnPollFunctor(
   };
 }
 
+// Optical-port-specific version.
+// Optical port messages are two level nested messages (DataResponse::T::U).
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc]: Same as here and below

File: stratum/hal/lib/common/yang_parse_tree_paths.cc (hunk 3 above)
Comment (pudelkoM) [reply stratum/hal/lib/common/yang_parse_tree_paths.cc]: It's not for the request, but for the response inner message.

Comment (Yi-Tseng) [reply stratum/hal/lib/common/yang_parse_tree_paths.cc]: `inner_msg_has_inner_msg_func`? (or maybe someting else?

File: stratum/hal/lib/common/yang_parse_tree_paths.cc (hunk 2 above)
Comment (pudelkoM) [reply stratum/hal/lib/common/yang_parse_tree_paths.cc]: Done. I left the comment about it being optical port specific to match the others.

**Pull request merged**